polaris-core/src/main/java/org/apache/polaris/core/policy/content/maintenance/MetadataCompactionPolicyContent.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.policy.content.maintenance;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Strings;
+import java.util.Set;
+import org.apache.polaris.core.policy.content.PolicyContentUtil;
+import org.apache.polaris.core.policy.validator.InvalidPolicyException;
+
+public class MetadataCompactionPolicyContent extends BaseMaintenancePolicyContent {
+  private static final String DEFAULT_POLICY_SCHEMA_VERSION = "2025-02-03";
+  private static final Set<String> POLICY_SCHEMA_VERSIONS = Set.of(DEFAULT_POLICY_SCHEMA_VERSION);
+
+  @JsonCreator
+  public MetadataCompactionPolicyContent(
+      @JsonProperty(value = "enable", required = true) boolean enable) {
+    super(enable);
+  }
+
+  public static MetadataCompactionPolicyContent fromString(String content) {
+    if (Strings.isNullOrEmpty(content)) {
+      throw new InvalidPolicyException("Policy is empty");
+    }
+
+    MetadataCompactionPolicyContent policy;
+    try {
+      policy = PolicyContentUtil.MAPPER.readValue(content, MetadataCompactionPolicyContent.class);
+    } catch (Exception e) {
+      throw new InvalidPolicyException(e);
+    }
+
+    if (policy == null) {
+      throw new InvalidPolicyException("Invalid policy: " + content);
+    }
+
+    if (Strings.isNullOrEmpty(policy.getVersion())) {
+      policy.setVersion(DEFAULT_POLICY_SCHEMA_VERSION);
+    }
```

Comment:
A lot of this logic looks duplicated, perhaps we can `super.validate` to reduce that

Reply:
Yeah, these duplicated code actually bothered me as well. I didn't refactor them eventually due to the reason that these schema just happened to have the same version number now. And some of schema just happened to have the same fields at this moment. I think it's fine to keep these duplication, so that we don't have to change the existing classes in case of version evolution and adding new policy schemas. It's kind of OCP. WDYT?

Reply:
I feel like a lot of these won't change as the policy schemas evolve, such as all this logic:

```
. . .

    if (policy == null) {
      throw new InvalidPolicyException("Invalid policy: " + content);
    }

    if (Strings.isNullOrEmpty(policy.getVersion())) {
      setPolicyVersion((getDefaultPolicySchemaVersion()));
    }

. . .

    if (!getPolicySchemaVersions.contains(getPolicyVersion())) {
      throw new InvalidPolicyException("Invalid policy version: " + getPolicyVersion());
    }
```

Reply:
Abstracted out the logic to a static method.